### ISSUE TITLE
Gradle plugin update

### ISSUE THREAD

**Issue by HipDragon** [serenity-bdd/serenity-core#2139]
To make life easier for Gradle users, I believe the Gradle plugin should be published.

It would also make sense that during this time the Gradle plugin gets updated to use the new API.

The documentation should also be updated to show both Groovy and Kotlin usage.

**Comment by wakaleo** [comment]
Great idea! Are you proposing to take on this task! I'll be more than happy to review your PR!

**Comment by HipDragon** [comment]
I thought about it, but if I did I would change the plugin over to Java code as recommended by Gradle. 

> In general, a plugin implemented using Java or Kotlin - [Custom Plugins](https://docs.gradle.org/current/userguide/custom_plugins.html)

If I did this work, how would the plugin get published?

**Comment by HipDragon** [comment]
@wakaleo I have started work on this, and would like to know a few things if you wouldn't mind giving me some answers.

I'm trying to figure out what all the directories are and there default values. I also noticed they are set by system environment variables, I was looking for something like the ThucydidesSystemProperty.SERENITY_PROJECT_KEY. Also thinking maybe it should be done as SerenitySystemProperty.SERENITY_PROJECT_KEY.

I also think this will need to be a separate project because it will depend on the Serenity-Core and allows users to pick their serenity version without the plugin forcing a version on them.
